app/controllers/serial_searches_controller.rb

```diff
@@ -8,9 +8,16 @@ def index
       @title = 'Serial Searches'
       per_page = params[:per_page] || 50
       page = params[:page] || 1
+
       @serial_searches_count = SerialSearch.count
       @ip_addresses_count = IpAddress.count
-      @serial_searches = SerialSearch.order(sort_column + ' ' + sort_direction).page(page).per(per_page)
+
+      if params[:query].present?
+        @serial_searches = SerialSearch.text_search(params[:query]).page(page).per(per_page)
```

Comment:
Are you suggesting to combine 16 and 18? The if was in case there wasn't a search I would get back all of the `SerialSearches`. I'm not exactly sure the `text_search` and `order` would work together.